compiler/qsc_frontend/src/resolve.rs

```diff
@@ -771,6 +807,21 @@ fn resolve(
         }
     }
 
+    if candidates.len() > 1 {
+        // If there are multiple candidates, remove unimplemented items.
```

Comment:
Why?

Reply:
Ah, the downsides of switching PRs... this question came up on the previous PR here: https://github.com/microsoft/qsharp/pull/785#discussion_r1365964792. Since it's come up twice, I think that's telling me I need to expand the comment to explain the scenario this is covering.